anta/tests/security.py

```diff
@@ -268,3 +270,68 @@ def test(self) -> None:
             self.result.is_failure(f"eAPI IPv6 ACL(s) not configured or active in vrf {self.inputs.vrf}: {not_configured_acl_list}")
         else:
             self.result.is_success()
+
+
+class VerifyCertificateStatus(AntaTest):
+    """
+    Verifies if SSL certificate details.
+    1. Verify if certificate is expired or about to expired
+    2. Certificate has correct name.
+    3. Certificate encryptions type and size.
+
+    Expected Results:
+        * success: The test will pass if expiry limit of certificate is grater then the threshold, has correct name, encryption and size.
+        * failure: The test will fail if certificate is expired or going to expired, has incorrect name, encryption and size.
+    """
+
+    name = "VerifyCertificateStatus"
+    description = "Verifies SSL certificate status."
+    categories = ["security"]
+    commands = [AntaCommand(command="show management security ssl certificate"), AntaCommand(command="show clock")]
+
+    class Input(AntaTest.Input):  # pylint: disable=missing-class-docstring
+        certificate: str
+        expiry_limit: int
+        subject_name: str
+        encryption: str
+        size: int
+        """Certificate, expiry limit, certificate subject name, encryption type and size"""
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        failure_msg = ""
+        certificate_output = self.instance_commands[0].json_output
+        clock_output = self.instance_commands[1].json_output
+
+        try:
+            # Collecting expiry time and current time.
+            certificate_output = certificate_output["certificates"][self.inputs.certificate]
+            expiry_time = certificate_output["notAfter"]
+            current_timestamp = clock_output["utcTime"]
+            day_difference = (datetime.datetime.utcfromtimestamp(expiry_time) - datetime.datetime.utcfromtimestamp(current_timestamp)).days
```

Comment:
Please use `datetime.datetime.fromtimestamp()`

https://docs.python.org/3/library/datetime.html#aware-and-naive-objects